### DISCUSSION TITLE
SHH key correct and working but still get "Permission Denied (publickey)"

### DISCUSSION THREAD

**Discussion by petermccutcheonsonos** [community/community#129720] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### GitHub Feature Area

Other

### Body

Hi,

I'm seeing this error message, "git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository."

Have followed this guide:
https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey

And all checks are correct, that is the SSH connection is working.

What could be other reasons that this is not working?

Terminal is in VScode.

Thanks

**Comment by queenofcorgis** [comment]
Hey there!  👋 

Thanks for posting in the GitHub Community, @petermccutcheonsonos ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category. The Accessibility category is a place for our community to discuss and provide feedback on the digital accessibility of GitHub products. Digital accessibility means that GitHub tools, and technologies, are designed and developed so that people with disabilities can use them.

I've gone ahead and moved this to the correct category for you. Good luck!

**Comment by maheshsehajpal** [comment]
Ensure that your SSH key is properly added to the ssh-agent and that the agent is running. Sometimes, even if the key is generated and added to GitHub, it might not be properly loaded into the ssh-agent.
You can add the key by running:
`ssh-add ~/.ssh/id_rsa  # or your specific key file`
